airflow/models/baseoperator.py

```diff
@@ -1718,8 +1615,11 @@ def __repr__(self) -> str:
 
     deps: Iterable[BaseTIDep] = attr.ib()
     operator_extra_links: Iterable['BaseOperatorLink'] = ()
-    params: Union[ParamsDict, dict] = attr.ib(factory=ParamsDict)
-    template_fields: Iterable[str] = attr.ib()
+    template_fields: Collection[str] = attr.ib()
+    template_ext: Collection[str] = attr.ib()
+
+    weight_rule: str = attr.ib()
+    priority_weight: int = attr.ib()
```

Comment:
You may need to add the properties to the serialize exclusion, or in some way adjust the serialization. I think right now if a mapped operator has either of these set it would appear twice in the json -- once at the top level and once in partial_kwargs.

Could you extend the tests/serialization/test_dag_serialization.py to check and assert this behaviour please?

Reply:
It looks like priority_weight isn't being serialized (based on the current failures) but it should be. Can be a follow up PR if you want.

Reply:
Actually I don’t think these should be serialised separately. Their states are already preserved in `partial_kwargs` and can be reconstructed appropriately on deserialisation in the current implementation.

Reply:
We have a different problem.

```
 'partial_kwargs': {'executor_config': {<Encoding.TYPE: '__type'>: <DagAttributeTypes.DICT: 'dict'>, <Encoding.VAR: '__var'>: {'dict': {<Encoding.TYPE: '__type'>: <DagAttributeTypes.DICT: 'dict'>, <Encoding.VAR: '__var'>: {'sub': 'value'}}}},
                    'priority_weight': 10},
 'priority_weight': 10,
```

It's in there twice, so you probaly want to exclude weight_rule, priority_weight and trigger_rule from being serialized, as they are already in partial kwargs.

Reply:
Done (also added a comment lamenting the situation)